test/integration/crd_validations_test.go

```diff
@@ -84,6 +86,27 @@ func TestCRDValidations(t *testing.T) {
 				require.ErrorContains(t, err, "spec.rules[0].port")
 			},
 		},
+		{
+			name: "service type=ExternalName with empty target port",
+			scenario: func(t *testing.T, cleaner *clusters.Cleaner, ns string) {
+				service := &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "service-name",
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeExternalName,
+						Ports: []corev1.ServicePort{
+							{
+								TargetPort: intstr.FromInt(0),
+							},
+						},
+						ExternalName: "example.com",
+					},
+				}
+				_, err := env.Cluster().Client().CoreV1().Services(ns).Create(ctx, service, metav1.CreateOptions{})
+				require.ErrorContains(t, err, "spec.ports[0].targetPort: Invalid value: 0")
+			},
+		},
```

Comment:
Eh, if it's Kubernetes behavior, and especially if it's not something we expect to vary between versions, no need to test it ourselves, so we may as well save the time.
```suggestion
```